### ISSUE TITLE
Document the legacy code

### ISSUE THREAD

**Issue by Technius** [xournalpp/xournalpp#1379]
There's a lot of undocumented legacy code. We should document it because it will be easier to:
* understand what's going on in the code
* contribute to the codebase through pull requests
* make changes to older code
* debug issues

We don't have to write extensive documentation, but each non-trivial function and method should have at least a one sentence description. Nor should our priority be to go back and document all of our old code right now. However, we should expect all _new_ code to have _some_ documentation; in addition, any refactoring in the legacy code should be accompanied by documentation changes.

**Comment by Febbe** [comment]
I would renounce the documentation for functions like ctors and dtors, operatorXX and really short good named functions (less equal than 10 to 15 lines of code).
The target should be to write self documenting code.

**Comment by andreasb242** [comment]
I fully agree with the documentation, in the Company we do this also. Even if the code also should be self documenting, this does simple not work, when different people are working on the same code.
